cylc/flow/job_runner_handlers/pbs.py

```diff
@@ -96,7 +96,10 @@ def format_directives(self, job_conf):
             f'{tokens["task"]}.{tokens["cycle"]}'
             f".{job_conf['workflow_name'].replace('/', '-')}"
         )
-        job_name_len_max = job_conf['platform']["job name length maximum"]
+        job_name_len_max = job_conf['platform'].get(
+            "job name length maximum",
+            self.JOB_NAME_LEN_MAX
+        )
```

Comment:
Thinking back, this wasn't working around the problem with `OrderedDictWithDefaults` not working right with `.get` was it?

Reply:
This is a revert of f12a0894ce88a18f3b3dae7193d83eff31107174. Would like to know why that commit did it this way in the first place

Reply:
There isn't any information on that.

Reply:
Judging from my description of the `OrderedDictWithDefaults.get()` problem here: https://github.com/cylc/cylc-flow/pull/4975#issuecomment-1188883694, I think this is fine.

Reply:
Can someone test this out with and without config defaults, for `localhost` and something-else, just to be safe.

Reply:
> Can someone test this out with and without config defaults, for `localhost` and something-else, just to be safe.

How do I test it out with localhost - it needs a pbs platform?

Reply:
You can take the `platform.get` code snippet out and put it anywhere you like, no need to test in-situ.

Reply:
```console
$ cylc config -i '[platforms][localhost]'
ssh command = ssh -oBatchMode=yes -oConnectTimeout=8 -oStrictHostKeyChecking=no
copyable environment variables = FCM_VERSION
submission polling intervals = PT1H
execution polling intervals = PT1H
clean job submission environment = True
```

```diff
diff --git a/cylc/flow/cfgspec/globalcfg.py b/cylc/flow/cfgspec/globalcfg.py
index 17ca5d98d..dbf2baed1 100644
--- a/cylc/flow/cfgspec/globalcfg.py
+++ b/cylc/flow/cfgspec/globalcfg.py
@@ -1891,6 +1891,8 @@ class GlobalConfig(ParsecConfig):
         # Flesh out with defaults
         self.expand()
 
+        breakpoint()
+
         self._no_platform_group_name_overlap()
         with suppress(KeyError):
             validate_platforms(self.sparse['platforms'])
```

```python
(Pdb) self.get(['platforms', 'localhost']).get('job runner')  # default
'background'
(Pdb) self.get(['platforms', 'localhost']).get('execution polling intervals')  # manually specified
[3600.0]
```